cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -275,6 +275,12 @@ public void upgrade() throws Exception {
   }
 
   @Override
+  public List<StreamSpecification> listStreams(NamespaceId namespaceId) throws Exception {
+    ensureAccess(namespaceId);
```

Comment:
@gokulavasan this seems wrong. this should instead filter the streams in the namespace and only return the ones to which the user has access.